drivers/wdm/audio/hdaudbus/businterface.cpp

```diff
@@ -363,10 +406,36 @@ HDA_PDOHandleQueryInterface(
         InterfaceHDA->FreeDmaBufferWithNotification = HDA_FreeDmaBufferWithNotification;
         InterfaceHDA->RegisterNotificationEvent = HDA_RegisterNotificationEvent;
         InterfaceHDA->UnregisterNotificationEvent = HDA_UnregisterNotificationEvent;
+        return STATUS_SUCCESS;
+    }
+    else if (IsEqualGUIDAligned(*IoStack->Parameters.QueryInterface.InterfaceType, GUID_HDAUDIO_BUS_INTERFACE_BDL))
+    {
+        InterfaceBDL = (PHDAUDIO_BUS_INTERFACE_BDL)IoStack->Parameters.QueryInterface.Interface;
+        InterfaceBDL->Version = IoStack->Parameters.QueryInterface.Version;
+        InterfaceBDL->Size = sizeof(HDAUDIO_BUS_INTERFACE_BDL);
+        InterfaceBDL->Context = DeviceExtension;
+        InterfaceBDL->InterfaceReference = HDA_InterfaceReference;
+        InterfaceBDL->InterfaceDereference = HDA_InterfaceDereference;
+
+        InterfaceBDL->TransferCodecVerbs = HDA_TransferCodecVerbs;
+        InterfaceBDL->AllocateCaptureDmaEngine = HDA_AllocateCaptureDmaEngine;
+        InterfaceBDL->AllocateRenderDmaEngine = HDA_AllocateRenderDmaEngine;
+        InterfaceBDL->ChangeBandwidthAllocation = HDA_ChangeBandwidthAllocation;
+        InterfaceBDL->FreeDmaEngine = HDA_FreeDmaEngine;
+        InterfaceBDL->SetDmaEngineState = HDA_SetDmaEngineState;
+        InterfaceBDL->GetWallClockRegister = HDA_GetWallClockRegister;
+        InterfaceBDL->GetLinkPositionRegister = HDA_GetLinkPositionRegister;
+        InterfaceBDL->RegisterEventCallback = HDA_RegisterEventCallback;
+        InterfaceBDL->UnregisterEventCallback = HDA_UnregisterEventCallback;
+        InterfaceBDL->GetDeviceInformation = HDA_GetDeviceInformation;
+        InterfaceBDL->GetResourceInformation = HDA_GetResourceInformation;
+
+        InterfaceBDL->AllocateContiguousDmaBuffer = HDA_AllocateContiguousDmaBuffer;
+        InterfaceBDL->FreeContiguousDmaBuffer = HDA_FreeContiguousDmaBuffer;
+        InterfaceBDL->SetupDmaEngineWithBdl = HDA_SetupDmaEngineWithBdl;
+        return STATUS_SUCCESS;
     }
 
-    // FIXME
-    // implement support for GUID_HDAUDIO_BUS_INTERFACE_BDL
-    UNIMPLEMENTED;
+    DPRINT1("Unsupported HDAUDIO GUID requested\n");
```

Comment:
Would it be worth to print the GUID too?

Reply:
Yes. But I doubt how to print it better. :thinking:

Reply:
Something like ba1ed89f?

Reply:
Ah, yes. Then this is only one way to print the passed GUID.